piptools/utils.py

```diff
@@ -522,3 +531,126 @@ def parse_requirements_from_wheel_metadata(
             markers=parts.markers,
             extras=parts.extras,
         )
+
+
+def callback_config_file_defaults(
+    ctx: click.Context, param: click.Parameter, value: str | None
+) -> str | None:
+    """
+    Returns the path to the config file with defaults being used, or `None` if no such file is
+    found.
+
+    Defaults for `click.Command` parameters should be override-able in a config file. `pip-tools`
+    will use the first file found, searching in this order: an explicitly given config file, a
+    `.pip-tools.toml`, a `pyproject.toml` file. Those files are searched for in the same directory
+    as the requirements input file.
+    """
+    if value is None:
+        config_file = select_config_file(ctx.params.get("src_files", ()))
+        if config_file is None:
+            return None
+    else:
+        config_file = Path(value)
+
+    try:
+        config = parse_config_file(config_file)
+    except OSError as e:
+        raise click.FileError(
+            filename=str(config_file), hint=f"Could not read '{config_file}': {e}"
+        )
+    except ValueError as e:
+        raise click.FileError(
+            filename=str(config_file), hint=f"Could not parse '{config_file}': {e}"
+        )
+
+    if not config:
+        return None
+
+    defaults: dict[str, Any] = ctx.default_map.copy() if ctx.default_map else {}
+    defaults.update(config)
+
+    ctx.default_map = defaults
+    return str(config_file)
+
+
+def select_config_file(src_files: tuple[str, ...]) -> Path | None:
+    """
+    Returns the config file to use for defaults given `src_files` provided.
+    """
+    if not src_files:
+        # If no src_files were specified, we consider the current directory the only candidate
+        candidate_dirs = [Path.cwd()]
+    else:
+        # Collect the candidate directories based on the src_file arguments provided
+        src_files_as_paths = [
+            Path(Path.cwd(), src_file).resolve() for src_file in src_files
```

Comment:
It took me a minute to get why you're feeding Path into Path... How about
```suggestion
            (Path.cwd() / src_file).resolve() for src_file in src_files
```

Reply:
Marking this thread as resolved.